src/verifications/verifications.ts

```diff
@@ -1469,43 +1552,31 @@ export class Verifications extends Logger {
     );
 
     // sign arguments for on-chain check
-    const issuerIdentity = await this.getIdentityForAccount(issuer);
-    const input = issuerIdentity.methods.addVerificationWithMetadata(
-      // uint256 _topic,
-      uint256VerificationName,
-      // uint256 _scheme,
-      '1',
-      // address _issuer,
-      sourceIdentity,
-      // bytes _signature,
-      signature,
-      // bytes _data,
-      verificationData,
-      // string _uri,
-      '',
-      // uint256 _expirationDate,
-      expirationDate,
-      // bytes32 _description,
-      ensFullNodeHash,
-      // bool _disableSubVerifications
-      disabelSubVerifications,
-    ).encodeABI();
+    const targetIdentityContract =
+      this.options.contractLoader.loadContract('VerificationHolder', targetIdentity);
 
-    // fetch nonce as late as possible
-    const nonce = executionNonce !== -1 ?
-      `${executionNonce}` : await this.getExecutionNonce(sourceIdentity, true);
-    // note that issuer is given for signing, as this ACCOUNT is used to sign the message
-    const signedTransactionInfo = await this.signPackedHash(
-      issuer, [sourceIdentity, nonce, targetIdentity, 0, input]);
+    const txInfo = await this.signTransaction(
+      targetIdentityContract,
+      'addVerificationWithMetadata',
+      { from: issuer, nonce: executionNonce },
+      uint256VerificationName,  // uint256 _topic,
+      '1',  // uint256 _scheme,
+      sourceIdentity,  // address _issuer,
+      signature,  // bytes _signature,
+      verificationData,  // bytes _data,
+      verificationDataUrl,  // string _uri,
+      expirationDate,  // uint256 _expirationDate,
+      ensFullNodeHash,  // bytes32 _description,
+      disabelSubVerifications,  // bool _disableSubVerifications
```

Comment:
maybe we can change the "disabelSubVerifications" to the correct phrase "disableSubVerifications"

Reply:
fixed